Sources/Network/URLSessionNetworkStack.swift

```diff
@@ -46,31 +63,28 @@ public extension Network {
             }
         }
 
-        public init(baseURL: URL, authenticationChallengeValidator: AuthenticationChallengeValidatorClosure? = nil) {
+        public init(baseURL: URL,
+                    authenticationChallengeValidator: AuthenticationChallengeValidatorClosure? = nil,
+                    authenticator: NetworkAuthenticator? = nil) {
             self.baseURL = baseURL
             self.authenticationChallengeValidator = authenticationChallengeValidator
+            self.authenticator = authenticator
         }
 
         public convenience init(configuration: Network.Configuration) {
             self.init(baseURL: configuration.baseURL,
-                      authenticationChallengeValidator: configuration.authenticationChallengeValidator)
+                      authenticationChallengeValidator: configuration.authenticationChallengeValidator,
+                      authenticator: configuration.authenticator)
         }
 
         @discardableResult
         public func fetch<R: NetworkResource>(resource: R,
                                              _ completion: @escaping Network.CompletionClosure) -> Cancelable {
-            guard let session = session else {
-                fatalError("🔥: session is `nil`! Forgot to 💉?")
-            }
+            let cancelableTask = CancelableTask()
 
-            let request = resource.toRequest(withBaseURL: baseURL)
+            performRequest(with: (resource: resource, cancelableTask: cancelableTask, completion: completion))
```

Comment:
And do what with that one? The task will be created only after the authenticate finishes and `fetch` has to return to the caller a `Cancelable`